packages/@react-aria/virtualizer/src/Virtualizer.tsx

```diff
@@ -98,48 +99,21 @@ interface VirtualizerOptions {
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 export function useVirtualizer<T extends object, V extends ReactNode, W>(props: VirtualizerOptions, state: VirtualizerState<T, V, W>, ref: RefObject<HTMLElement | null>) {
   let {isLoading, onLoadMore} = props;
-  let {setVisibleRect, virtualizer} = state;
+  let {setVisibleRect} = state;
+  let scrollOffset = ref.current?.clientHeight;
+
+  // TODO: The ref previously provided was the table's ref. That ref didn't actually do anything in the old implementation
+  // but now the ref provided to useVirtualizer MUST be the scrollable ref so its a breaking change kinda?
+  // TODO: Also, now that we use ref.currnet.clientHeight/scrollHeight/etc instead of relying on the rect to figure out if we should load more
+  // this will break peoples tests since they need to mock scrollHeight to simulate the virtualizer's total contents height...
+  let {scrollViewProps: {onScroll}} = useLoadOnScroll({isLoading, onLoadMore, scrollOffset}, ref);
```

Comment:
Maybe a breaking change to have useVirtualizer use useLoadOnScroll for the above reasons? Open to discussion if useVirtualizer should not use useLoadOnScroll

Reply:
Should virtualizer even be responsible for this at all? Maybe we should just call `useLoadOnScroll` directly from within collection components that need it?

Reply:
For posterity: since we are bumping virtualizer package, will try and get rid of useVirtualizer entirely and have RSP components call useLoadOnScroll directly. See https://github.com/adobe/react-spectrum/pull/6494#discussion_r1659422726 for some issues I ran into when trying to do this, will hopefully resolve after break